src/k8s/pkg/component/dns.go

```diff
@@ -0,0 +1,136 @@
+package component
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/utils/k8s"
+)
+
+func EnableDNSComponent(s snap.Snap, clusterDomain, serviceIP string, upstreamNameservers []string) error {
+	manager, err := NewManager(s)
+	if err != nil {
+		return fmt.Errorf("failed to get component manager: %w", err)
+	}
+
+	upstreamNameserver := "/etc/resolv.conf"
+	if clusterDomain == "" {
+		clusterDomain = "cluster.local"
+	}
+
+	if len(upstreamNameservers) > 0 {
+		upstreamNameserver = strings.Join(upstreamNameservers, " ")
+	}
+
+	values := map[string]any{
```

Comment:
Can we also set `service.name=coredns` and `deployment.name=coredns`? Will reduce some of the stutter in the deployed names.

- https://github.com/coredns/helm/blob/master/charts/coredns/values.yaml#L60
- https://github.com/coredns/helm/blob/master/charts/coredns/values.yaml#L376